pymc3/sampling.py

```diff
@@ -412,6 +422,25 @@ def sample(
     if not isinstance(random_seed, Iterable):
         raise TypeError("Invalid value for `random_seed`. Must be tuple, list or int")
 
+    if not discard_tuned_samples and not return_inferencedata:
+        warnings.warn(
+            "Tuning samples will be included in the returned `MultiTrace` object. "
+            "To avoid complications with downstream analysis, you should slice them out:\n"
+            "trace_warmup = trace[: -trace.report.n_draws]\n"
+            "trace_posterior = trace[-trace.report.n_draws :]\n"
+            "Alternatively, you can use InferenceData: `pm.sample(..., return_inferencedata=True)`",
```

Comment:
Actually, I'm thinking that we should remove the advice about slicing and just nudge users towards using ID, which takes care of warmup and posterior groups automatically, don't you think? 
This will avoid complications in downstream analysis _and_ help users get used to ID, which is probably gonna be the new default anyway. 
cc @OriolAbril

Reply:
I made the change. However, I'd like to remind us that the warmup groups are still in a hidden attribute of the `InferenceData`